src/lib.rs

```diff
@@ -488,22 +497,46 @@ mod test {
     assert_eq!(data, xor(&xor(&data,&pad), &pad));
   }
 
+
+
   #[test]
-  fn check_write() {
+  fn check_write_and_read() {
     let mut my_storage = MyStorage::new();
+    let size = 3 as u64;
```

Comment:
you define it as an immutable variable, but what is the value of that if the code explicitly depends on this value being 3.  This btw is the answer to the rhetorical question below.